include/swift/AST/Decl.h

```diff
@@ -819,7 +819,11 @@ class alignas(1 << DeclAlignInBits) Decl {
 
   // Make vanilla new/delete illegal for Decls.
   void *operator new(size_t Bytes) = delete;
+
+  // Work around MSVC error: attempting to reference a deleted function.
```

Comment:
I get the following error

> error C2280: 'void *swift::AST::ModuleDecl::__delDtor(unsigned int)': attempting to reference a deleted function 

The problem seems to be that MSVC generates this `__delDtor` intrinsic that I don't seem to have control over.